pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -432,8 +432,8 @@ func handleRule(action workloadapi.Action, rule *v1beta1.Rule) []*workloadapi.Ru
 		rules = append(rules, &workloadapi.Rules{Matches: fromMatches})
 	}
 	for _, when := range rule.When {
-		l7 := l4WhenAttributes.Contains(when.Key)
-		if action == workloadapi.Action_ALLOW && !l7 {
+		l4 := l4WhenAttributes.Contains(when.Key)
+		if action == workloadapi.Action_ALLOW && !l4 {
```

Comment:
can you please change the comment below as well?

Reply:
the comment is right. I think the implicit meaning of `!l4` here is "it's l7".

Reply:
Ah I see. Misread the code